### DISCUSSION TITLE
FastDDS v3 TypeObject and its dependencies requirement to create a Dynamic type at remote side

### DISCUSSION THREAD

**Discussion by arnabgho81** [eProsima/Fast-DDS#5221] [Q&A] [answered]
There are no examples for Typelookup service. I want to know how do I deserialize a topic sample whos type is dependent of other types. Do I only need the typeobject for that type only and register or need all the dependent types and its dependent and register.
Also When will the FastDDS v3 Recorder and Replay service be available.

**Comment by juanlofer-eprosima** [comment, marked as answer]
Hi @arnabgho81,

If using Fast-DDS 2.x , you may have a look at [this example](https://github.com/eProsima/Fast-DDS/tree/v2.14.3/examples/cpp/dds/TypeLookupService) which uses Typelookup service. However, this version does not support complex types with dependencies as it is your case, so I encourage you to use Fast-DDS 3.x instead where the xtypes module has completely been refactored. You may have a look at [this example](https://github.com/eProsima/Fast-DDS/tree/v3.0.0/examples/cpp/xtypes) showing how to send and receive xtypes via TypeLookup service.

Answering your question, no it is not required to manually register dependencies. If the type support is generated with Fast-DDS-Gen from an idl file (in which both the "main" type and its dependencies are specified), it suffices to register (`DomainParticipant::register_type`) the type for which data is to be published and internally Fast-DDS should take care of registering all dependencies. TypeLookup service will also take care of sending the "main" type together with all its dependencies, so nothing needs to be done in the receiver side; once it is notified via a discovery callback that the "main" type is received, it should be the case that all its dependencies were received and registered as well.

Please feel free to check the following [section](https://fast-dds.docs.eprosima.com/en/v3.0.0/fastdds/xtypes/discovery_matching.html#xtypes-discovery-matching) for more details on remote types discovery in the new xtypes implementation. For your information, we also introduced a couple of utility functions meant to ease the interaction with dynamic types and data: [serialization utilities](https://fast-dds.docs.eprosima.com/en/v3.0.0/fastdds/xtypes/type_serializing.html)

Regarding your last question, the next DDS-Record-Replay release (working with Fast-DDS 3.0.1) is scheduled for next week.

Best regards.

**Reply by arnabgho81** [reply]
Thanks for the prompt reply. the information was useful to get the type object and dynamic types working.

**Reply by arnabgho81** [reply]
May be one more query, I was going through the fastdds code I am not able to understand who makes the call to TypeLookupManager::get_types that initiates the type discovery process.

**Reply by juanlofer-eprosima** [reply]
This method is called asynchronously by the TypeLookup service, in particular from `TypeLookupReplyListener::check_get_type_dependencies_reply`.
